bot/cogs/bot.py

```diff
@@ -100,81 +100,229 @@ def __init__(self, bot: AutoShardedBot):
 
-                # Strip again to remove the whitespace(s) left before the code
-                # If the msg looked like "Python <code>" before removing Python
-                content = content.strip()
-                log.trace(f"Returning message.\n\n{content}\n\n")
-                return content
+        Tries to strip out REPL Python code out of msg and returns the stripped msg.
+        """
+        final = ""
+        for line in msg.splitlines(keepends=True):
+            if line.startswith(">>>") or line.startswith("..."):
+                final += line[4:]
+        log.trace(f"Formatted: \n\n{msg}\n\n to \n\n{final}\n\n")
+        if not final:
+            log.debug(f"Found no REPL code in \n\n{msg}\n\n")
+            return msg, False
+        else:
+            log.debug(f"Found REPL code in \n\n{msg}\n\n")
+            return final.rstrip(), True
 
     async def on_message(self, msg: Message):
-        if not msg.author.bot:
-            if msg.channel.id in self.channel_cooldowns:
-                on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
-                if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
-                    try:
-                        content = self.codeblock_stripping(msg.content)
-                        if not content:
+        if msg.channel.id in self.channel_cooldowns and not msg.author.bot and len(msg.content.splitlines()) > 3:
+            on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
+            if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
+                try:
+                    not_backticks = ["'''", '"""', "´´´", "‘‘‘", "’’’", "′′′", "“““", "”””", "″″″", "〃〃〃"]
+                    bad_ticks = msg.content[:3] in not_backticks
+                    if bad_ticks:
+                        ticks = msg.content[:3]
+                        content = self.codeblock_stripping(f"```{msg.content[3:-3]}```", True)
+                        if content is None:
+                            return
+
+                        content, repl_code = content
+
+                        if len(content) == 2:
+                            content = content[1]
+                        else:
+                            content = content[0]
+
+                        space_left = 204
+                        if len(content) >= space_left:
+                            current_length = 0
+                            lines_walked = 0
+                            for line in content.splitlines(keepends=True):
+                                if current_length+len(line) > space_left or lines_walked == 10:
+                                    break
+                                current_length += len(line)
+                                lines_walked += 1
+                            content = content[:current_length] + "#..."
+
+                        howto = ("It looks like you are trying to paste code into this channel.\n\n"
```

Comment:
Can we change these indents pls
```py
howto = (
    "It looks like you are trying to paste code into this channel.\n\n"
    "You seem to be using the wrong symbols to indicate where the"
    " codeblock should start. The correct symbols would be \`\`\`, not"
    f" `{ticks}`.\n\n**Here is an example of how it should look:**\n"
    f"\`\`\`python\n{content}\n\`\`\`\n\n**This will result in the "
    f"following:**\n```python\n{content}\n```"
)
```